contrib/CMakeLists.txt

```diff
@@ -25,6 +25,8 @@ function(add_contrib cmake_folder)
         endif()
 
         file(GLOB contrib_files "${base_folder}/*")
+        # remove .git from contrib_files
```

Comment:
I don't get it. What bad things will happen if the `.git` directory is not removed from a submodule directory? (which was the situation before your PR)

Reply:
When the network environment is unstable, git may encounter errors while downloading a submodule, potentially leaving only a .git file in the submodule directory.
```
jiangyinzuo@kiwi:/usr3/jiangyinzuo_data/ClickHouse/contrib/mariadb-connector-c$ cat .git
gitdir: ../../.git/modules/contrib/mariadb-connector-c
jiangyinzuo@kiwi:/usr3/jiangyinzuo_data/ClickHouse/contrib/mariadb-connector-c$ ls -a
.  ..  .git
```

In this situation, the submodule is equivalent to an empty repository and should be considered as not having been successfully initialized.

Without this patch, the result of `file(GLOB contrib_files "${base_folder}/*")` is ["path/to/contrib/submodule/.git"], which is not empty, so `if (NOT contrib_files)` is true, the following statement will not be executed.
https://github.com/ClickHouse/ClickHouse/blob/02237043c00fdcd2f27832094080e6430a3b5fb2/contrib/CMakeLists.txt#L30-L36

With this patch, we can remind users to pay attention to updating the submodule.

Reply:
I guess it is sane to assume that at the time `cmake` runs, all submodules are actually checked out. The situation you described (only a `.git` file/directory exists) seems quite specific, there may be hundreds of other things that can to wrong during checkout, including stuff like partial checkouts, wrong permissions, the device was unmounted etc. It is infeasible to understand what could exactly go wrong, and therefore we should also not try to handle one out of the infinitely many error cases.

What I find quite convenient is showing the Git status in the shell prompt, see e.g. here: https://github.com/rschu1ze/dotfiles/blob/main/.bashrc#L117